### PR TITLE
WIP: Improve types

### PR DESCRIPTION
**Pull request by mkslanc**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Pull Request Checklist:
* [x] No backwards incompatible changes were made to Ace's public interface which is defined through the main typings file (`ace.d.ts`) and its references:
    * https://github.com/ajaxorg/ace/blob/master/ace.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-modes.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-extensions.d.ts

### CODE REVIEW

**Comment by akoreman** [comment]
Hey @mkslanc, thanks for the PR! I think it's a step in the right direction, is there anything blocking it from being ready for review?

**Comment by mkslanc** [comment]
Hey @akoreman . I just waited until #5690 was merges, as it has some type intersections with that PR.

**Review by akoreman — APPROVED** [review]

File: ace-internal.d.ts [hunk 1]
```diff
@@ -1554,7 +1564,3 @@ declare module "./src/mouse/default_gutter_handler" {
     }
 }
 
-declare module "./src/lib/keys" {
```
Comment (akoreman) [inline ace-internal.d.ts:1557]: Missed this during the initial review, what was the reason for removing the type declaration of `lib/keys`?

Comment (mkslanc) [reply ace-internal.d.ts:1557]: Actually that was wrong, It exported only one function without repeating structure for this module, which is wild mix of 
```javascript
oop.mixin(exports, Keys);
exports.default = exports;
exports.keyCodeToString
```

**Pull request merged**